main.tf

```diff
@@ -1,13 +1,180 @@
+    organization = var.organization_name
+  }
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  basic_constraints = {
+    ca = true
   }
+
+  allowed_uses = [
+    "crl_signing",
+    "cert_signing",
+  ]
+
+  context = module.this.context
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+module "self_signed_cert_root" {
+  source = "cloudposse/ssm-tls-self-signed-cert/aws"
+
+  name = "self-signed-cert-root"
+
+  subject = {
+    common_name  = var.root_common_name ? var.root_common_name : module.this.id
+    organization = var.organization_name
+  }
+
+  basic_constraints = {
+    ca = false
+  }
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth"
+  ]
+
+  context = module.this.context
+}
+
+module "self_signed_cert_server" {
+  source = "cloudposse/ssm-tls-self-signed-cert/aws"
+
+  name = "self-signed-cert-server"
+
+  subject = {
+    common_name  = var.server_common_name ? var.server_common_name : module.this.id
+    organization = var.organization_name
+  }
+
+  basic_constraints = {
+    ca = false
+  }
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth"
+  ]
+
+  context = module.this.context
+}
+
+module "cloudwatch_log" {
+  source = "cloudposse/cloudwatch-logs/aws"
+
+  enabled = var.logging_enabled
+
+  stream_names = [var.logging_stream_name]
+
+  context = module.this.context
+}
+
+resource "aws_ec2_client_vpn_endpoint" "default" {
+  description            = module.this.id
+  server_certificate_arn = module.self_signed_cert_server.certificate_arn
+  client_cidr_block      = var.client_cidr
+
+  dynamic "authentication_options" {
+    for_each = var.authentication_options
+    content {
+      type                       = each.key
+      root_certificate_chain_arn = each.value
+    }
+  }
+
+  connection_log_options {
+    enabled               = var.logging_enabled
+    cloudwatch_log_group  = var.logging_enabled ? module.cloudwatch_log.log_group_name : null
+    cloudwatch_log_stream = var.logging_enabled ? var.logging_stream_name : null
+  }
+
+  tags = module.this.tags
+}
+
+resource "aws_ec2_client_vpn_authorization_rule" "internet_rule" {
+  count = local.enabled && var.internet_access_enabled ? 1 : 0
+
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
+  target_network_cidr    = "0.0.0.0/0"
+  authorize_all_groups   = true
+}
+
+module "vpn_security_group" {
+  source = "cloudposse/security-group/aws"
+
+  attributes = ["simple"]
+  rules = [
+    {
+      key         = "vpn-self"
+      type        = "ingress"
+      from_port   = 0
+      to_port     = 0
+      protocol    = "-1"
+      cidr_blocks = ["0.0.0.0/0"]
+      description = "Allow self access only by default"
+      self        = true
+    },
+  ]
+
+  vpc_id = module.vpc.vpc_id
+
+  context = module.this.context
+}
```

Comment:
This doesn't resemble how we use the security group module elsewhere. The objective security group module is ot expose an interface that makes it easier for security-conscious companies to fine-tune the rules. Thus we want to expose rules as variables.

Reply:
In this case, it is the default security group for the VPC. We want the rules fixed such that it is a minimum requirement.